### PR TITLE
Add procedure for pods not starting

### PR DESCRIPTION
**Pull request by bklei**
## Summary and Scope

Need a procedure for how to recover from pods stuck in containercreating/init.

## Issues and Related PRs

* Resolves [CASMPET-5582](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-5582)

## Testing

Tested this procedure on frigg.

### Tested on:

  * `frigg`

### Test description:

Tested this procedure on frigg.

## Risks and Mitigations

Low

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Prerequisites
+
+The `kubectl get pod` command returns pods that seem to be stuck in the `Init` or `ContainerCreating` state.
+
+## Identify the node in question
+
+1. Run the `kubectl get pod -o wide` command to identify the node where the pod isn't starting.
+
+    ```bash
+    ncn-w001# kubectl get pod -A -o wide | grep 'Init\|ContainerCreating'
```
Comment (cdelatte-hpe) [inline troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:14]: alternatively you could use egrep 'Init|ContainerCreating'

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   ```bash
+   ncn-w002# systemctl restart kubelet
+   ```
+
+1. Restart the containerd service.
+
+   ```bash
+   ncn-w002# systemctl restart containerd
+   ```
+
```
Comment (cdelatte-hpe) [inline troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:36]: should there be a systemctl status or is_active check to make sure it restarted?

**Review by cdelatte-hpe — COMMENTED** [review]
a couple of comments.  not blockers, just something that might make sense to have

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md (hunk 1 above)
Comment (bklei) [reply troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:14]: sounds good -- will change

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md (hunk 2 above)
Comment (bklei) [reply troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:36]: great comment, will fix

**Review by cdelatte-hpe — APPROVED** [review]

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Troubleshoot Kubernetes Pods Not Starting
+
+Use this procedure to check if Kubernetes pods get scheduled on an NCN, but don't eventually reach the `Running` state.
```
Comment (mharding-hpe) [inline troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:3]:
```suggestion
Use this procedure to check if Kubernetes pods get scheduled on an NCN, but do not eventually reach the `Running` state.
```

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    services  gitea-vcs-65c98746b-jk5v7   0/2  ContainerCreating  0  2d3h    10.47.0.104  ncn-w002  <none>  <none>
+    ```
+
+    In the above example, `ncn-w002` is the node that may need attention.
+
+## Recovery Steps
+
+Execute the following steps the node in determined in the previous step.
+
+1. Restart the kubelet service.
```
Comment (mharding-hpe) [inline troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:25]:
```suggestion
1. Restart the `kubelet` service.
```

File: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+   ncn-w002# systemctl status kubelet
+   ```
+
+1. Restart the containerd service.
+
+   ```bash
+   ncn-w002# systemctl restart containerd
+   ```
+
+   Now ensure containerd is running:
```
Comment (mharding-hpe) [inline troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md:43]:
```suggestion
1. Ensure that `containerd` is running.
```

**Pull request merged**